tests/src/junit/java/net/neoforged/neoforge/unittest/transfer/EnergyTests.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.unittest.transfer;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.mojang.serialization.Codec;
+import java.util.function.Supplier;
+import net.minecraft.core.NonNullList;
+import net.minecraft.core.component.DataComponentType;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.network.codec.ByteBufCodecs;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.neoforged.bus.api.IEventBus;
+import net.neoforged.fml.common.Mod;
+import net.neoforged.neoforge.registries.DeferredRegister;
+import net.neoforged.neoforge.transfer.access.ItemAccess;
+import net.neoforged.neoforge.transfer.energy.EmptyEnergyHandler;
+import net.neoforged.neoforge.transfer.energy.EnergyHandler;
+import net.neoforged.neoforge.transfer.energy.ItemAccessEnergyHandler;
+import net.neoforged.neoforge.transfer.energy.SimpleEnergyHandler;
+import net.neoforged.neoforge.transfer.item.ItemResource;
+import net.neoforged.neoforge.transfer.item.ItemStacksResourceHandler;
+import net.neoforged.neoforge.transfer.transaction.Transaction;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+import org.junit.jupiter.api.Test;
+
+class EnergyTests {
+    @Test
+    public void testEmptyEnergyHandler() {
+        try (Transaction transaction = Transaction.open(null)) {
+            ensureEmpty(EmptyEnergyHandler.INSTANCE, transaction);
+        }
+    }
+
+    @Test
+    void testSimpleEnergyHandler() {
+        var simpleHandler = new SimpleEnergyHandler(100, 5, 10);
+        assertEquals(0, simpleHandler.getAmountAsLong());
+
+        try (Transaction transaction = Transaction.open(null)) {
+            assertEquals(5, simpleHandler.insert(100, transaction));
+            assertEquals(3, simpleHandler.insert(3, transaction));
+            assertEquals(8, simpleHandler.getAmountAsLong());
+        }
+
+        assertEquals(0, simpleHandler.getAmountAsLong());
+
+        try (Transaction transaction = Transaction.open(null)) {
+            assertEquals(5, simpleHandler.insert(100, transaction));
+            assertEquals(3, simpleHandler.insert(3, transaction));
+            assertEquals(8, simpleHandler.getAmountAsLong());
+
+            transaction.commit();
+        }
+
+        assertEquals(8, simpleHandler.getAmountAsLong());
+    }
+
+    @Test
+    void testItemAccessEnergyHandler() {
+        // Starting items: 2 diamonds. Diamond will be considered the energy container here.
+        var startingItems = NonNullList.of(ItemStack.EMPTY, new ItemStack(Items.DIAMOND, 2));
+        var handler = new ItemStacksResourceHandler(startingItems) {
+            @Override
+            protected int getCapacity(int index, ItemResource resource) {
```

Comment:
Is there any reason for this override and making an anonymous class? (Especially in the context of just energy on the corresponding stack)